### ISSUE TITLE
Add required CTP payment custom types

### ISSUE THREAD

**Issue by floriansattler** [commercetools/commercetools-paypal-plus-integration#4]
Add automated builder on service run which verifies the required custom types exists and have proper custom fields/types
#40

- if the custom type/filed does not exists - create one
- if a custom filed with the same name has different type - throw/log an exception and shut down the service

**Comment by andrii-kovalenko-ct** [comment]
example for tests:
```json
{
  "key": "payment-paypal",
  "name": {
    "en": "payment-paypal"
  },
  "resourceTypeIds": [
    "payment"
  ],
  "fieldDefinitions": [
    {
      "name": "languageCode",
      "label": {
        "en": "languageCode"
      },
      "required": true,
      "type": {
        "name": "String"
      },
      "inputHint": "SingleLine"
    },
    {
      "name": "reference",
      "label": {
        "en": "reference"
      },
      "required": true,
      "type": {
        "name": "String"
      },
      "inputHint": "SingleLine"
    },
    {
      "name": "referenceText",
      "label": {
        "en": "referenceText"
      },
      "required": false,
      "type": {
        "name": "String"
      },
      "inputHint": "MultiLine"
    },
    {
      "name": "redirectUrl",
      "label": {
        "en": "redirectUrl"
      },
      "required": false,
      "type": {
        "name": "String"
      },
      "inputHint": "SingleLine"
    },
    {
      "name": "successUrl",
      "label": {
        "en": "successUrl"
      },
      "required": false,
      "type": {
        "name": "String"
      },
      "inputHint": "SingleLine"
    },
    {
      "name": "errorUrl",
      "label": {
        "en": "errorUrl"
      },
      "required": false,
      "type": {
        "name": "String"
      },
      "inputHint": "SingleLine"
    },
    {
      "name": "cancelUrl",
      "label": {
        "en": "cancelUrl"
      },
      "required": false,
      "type": {
        "name": "String"
      },
      "inputHint": "SingleLine"
    }
  ]
}
```

**Comment by andrii-kovalenko-ct** [comment]
@butenkor 
for the custom fields name should we try to use the same names, as for Payone, or better to reflect Paypal Plus specific names?

For example, in payone we have: _successUrl_, _redirectUrl_, _cancelUrl_, _errorUrl_

In Paypal Plus we have: _returnUrl_ (like successUrl), _cancelUrl_, _approvalUrl_ (like redirectUrl)

**Comment by andrii-kovalenko-ct** [comment]
current custom types from `paypalplus-integration-test`:
```json
[
  {
    "key": "paypal-plus-interaction-request",
    "name": {
      "en": "paypal-plus-interaction-request"
    },
    "resourceTypeIds": [
      "payment-interface-interaction"
    ],
    "fieldDefinitions": [
      {
        "name": "request",
        "label": {
          "en": "request"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "MultiLine"
      },
      {
        "name": "timestamp",
        "label": {
          "en": "timestamp"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "transactionId",
        "label": {
          "en": "transactionId"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      }
    ]
  },
  {
    "key": "paypal-plus-interaction-response",
    "name": {
      "en": "paypal-plus-interaction-response"
    },
    "resourceTypeIds": [
      "payment-interface-interaction"
    ],
    "fieldDefinitions": [
      {
        "name": "timestamp",
        "label": {
          "en": "timestamp"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "transactionId",
        "label": {
          "en": "transactionId"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "response",
        "label": {
          "en": "response"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "MultiLine"
      }
    ]
  },
  {
    "key": "paypal-plus-interaction-notification",
    "name": {
      "en": "paypal-plus-interaction-notification"
    },
    "resourceTypeIds": [
      "payment-interface-interaction"
    ],
    "fieldDefinitions": [
      {
        "name": "timestamp",
        "label": {
          "en": "timestamp"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "notification",
        "label": {
          "en": "notification"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "MultiLine"
      }
    ]
  },
  {
    "key": "payment-paypal",
    "name": {
      "en": "payment-paypal"
    },
    "resourceTypeIds": [
      "payment"
    ],
    "fieldDefinitions": [
      {
        "name": "languageCode",
        "label": {
          "en": "languageCode"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "reference",
        "label": {
          "en": "reference"
        },
        "required": true,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "referenceText",
        "label": {
          "en": "referenceText"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "MultiLine"
      },
      {
        "name": "redirectUrl",
        "label": {
          "en": "redirectUrl"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "successUrl",
        "label": {
          "en": "successUrl"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "errorUrl",
        "label": {
          "en": "errorUrl"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "cancelUrl",
        "label": {
          "en": "cancelUrl"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "approvalUrl",
        "label": {
          "de": "approvalUrl",
          "en": "approvalUrl"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "payerId",
        "label": {
          "de": "payerId",
          "en": "payerId"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "paidToAccountBankName",
        "label": {
          "en": "paidToAccountBankName"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "paidToAccountHolderName",
        "label": {
          "en": "paidToAccountHolderName"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "paidToIBAN",
        "label": {
          "en": "paidToIBAN"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "paidToBIC",
        "label": {
          "en": "paidToBIC"
        },
        "required": false,
        "type": {
          "name": "String"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "paymentDueDate",
        "label": {
          "en": "dueDate"
        },
        "required": false,
        "type": {
          "name": "Date"
        },
        "inputHint": "SingleLine"
      },
      {
        "name": "amount",
        "label": {
          "en": "amount"
        },
        "required": false,
        "type": {
          "name": "Money"
        },
        "inputHint": "SingleLine"
      }
    ]
  }
]
```

**Comment by andrii-kovalenko-ct** [comment]
will be resolved as a part of #40

**Comment by andrii-kovalenko-ct** [comment]
resolved by https://github.com/commercetools/commercetools-paypal-plus-integration/pull/107
